querygen/querygen.go

```diff
@@ -180,3 +180,28 @@ func GenReleaseCompletedQuery(index string) string {
 		String()
 	return q
 }
+
+// GenPullRequestCreatedQuery returns a Splunk query for generating Segment events
+// whenever a Pull request is created in the users GitHub repository.
+func GenPullRequestCreatedQuery(index string) string {
+	q, _ := NewUserJourneyQuery(index).
+		WithPredicate(
+			`verb=update `+
+				`"responseStatus.code"=200 `+
+				`"objectRef.apiGroup"="appstudio.redhat.com" `+
+				`"objectRef.resource"="components" `+
+				`"user.username"="system:serviceaccount:build-service:build-service-controller-manager" `+
+				`"responseObject.metadata.annotations.build.appstudio.openshift.io/status"="*pac*" `+
+				`(NOT "responseObject.metadata.annotations.build.appstudio.openshift.io/request"="*")`,
+		).
+		WithCommands(
+			`spath input="responseObject.metadata.annotations.build.appstudio.openshift.io/status", path=pac.state output=build_status.pac.state`,
+			`search "build_status.pac.state"="enabled"`,
+			`spath input="responseObject.metadata.annotations.build.appstudio.openshift.io/status", path=pac.merge-url output=build_status.pac.merge-url`,
+			`dedup build_status.pac.merge-url sortby +_time`,
+		).
+		WithEventExpr(`"Pull request created"`).
+		WithFields("namespace", "name", "application", "component", "merge_url").
```

Comment:
Since this is querying component objects, we should also include the src_url and src_revision fields that we recently defined so that we can cross-reference component events for the same repo.